lib/compress/zstd_compress.c

```diff
@@ -2287,12 +2299,19 @@ static size_t ZSTD_compressBlock_internal(ZSTD_CCtx* zc,
             zc->entropyWorkspace, HUF_WORKSPACE_SIZE /* statically allocated in resetCCtx */,
             zc->bmi2);
 
+    if (cSize < 10 && ZSTD_isRLE(ip, srcSize)) {
```

Comment:
1) I guess it means that the threshold to determine that a block _might_ be RLE is 10 bytes ?
In which case, give a name to this constant, it's useful for code documentation.

2) A previous bootcamper tried to generate statistics on this topic.
His results suggested that the limit should be higher, because many RLE-possible blocks ended up being larger than example. His suggestion was rather in the ~25 bytes.
The question of the threshold is probably worth exploring.

Reply:
The `dictionary_round_trip` regression test suggests ~25 bytes as well.

However, it also has an example where an RLE block is 423 bytes. In that case literal compression is disabled. So unless we don't use a heuristic, we'll never get it exactly right.

A middle ground could be to use a larger than necessary heuristic, but spend a little bit of time optimizing `ZSTD_isRLE()` so we can afford to run it more often.

Reply:
At this stage, I believe we are fine missing a few outliers.
Catching the remaining ones could be seen as a later refinement, if it ever becomes a topic.